analyzer/src/main/kotlin/managers/Carthage.kt

```diff
@@ -101,8 +101,8 @@ class Carthage(
         val vcsHost = VcsHost.toVcsHost(URI(normalizedVcsUrl))
 
         return ProjectInfo(
-            user = vcsHost?.getUserOrOrganization(normalizedVcsUrl),
-            project = vcsHost?.getProject(normalizedVcsUrl),
+            namespace = vcsHost?.getUserOrOrganization(normalizedVcsUrl),
```

Comment:
This change lead me to the topic of unique-ness of identifiers.
As we do require the identifiers to be unique, I believe it'd be better to use the path relative to the analyzer root.
IIRC this was done in other package managers as well.

out-of-scope: to make thinks even more unique, would it make sense to include also the VCS host in the identifier?

Reply:
@MarcelBochtler I believe @fviernau is referring to how we already do it e.g. for Pip, see

https://github.com/oss-review-toolkit/ort/blob/e0bc11c24525e6c7cf12972e695f69a3311f9574/analyzer/src/main/kotlin/managers/Pip.kt#L315-L317

and I agree aligning on this makes sense.

> out-of-scope: to make thinks even more unique, would it make sense to include also the VCS host in the identifier?

The question is whether it really should be *that* unique. I.e., maybe it's beneficial to treat Git mirrors the same? But let's continue that discussion elsewhere.

Reply:
Did you comment the wrong commit here?
In 3e9b2a98cd630a38ed7cc1c84e85b604a0d53ce8 I see that there is an issue:
```
        return ProjectInfo(
            namespace = vcsHost?.getUserOrOrganization(normalizedVcsUrl),
            projectName = vcsHost?.getProject(normalizedVcsUrl) ?: workingDir.path.substringAfterLast("/"),
            revision = vcsInfo.revision
        )
```

We won't have a `UserOrOrganization` and the `projectName` would be retrieved from the working directory.
This would be an issue in the following folder layout:
```
❯ tree
.
├── foo
│   └── myproject
│       └── Cartfile.resolved
└── myproject
    └── Cartfile.resolved
```
and therefore lead to the identifier `Carthage::myproject:<revision>`

Reply:
For above example my proposal from above applied would give
- `Carthage::myproject:<revision>`
- `Carthage::foo/myproject:<revision>`

Reply:
Fixed it. For this tree:
```
.
├── myproject
│   ├── Cartfile.resolved
└── test
    └── myproject
        └── Cartfile.resolved
```

I'm getting:
`"Carthage::myproject:<revision>"`
`"Carthage::test/myproject:<revision>"`